docs/rst/tutorials/micro/memory_management/memory_management.rst

```diff
@@ -0,0 +1,430 @@
+
+In the default configuration, micro-ROS Client will not be able to create more entities than the ones specified above.
+If dynamic memory usage is allowed, by means of the following flag, the user can allow on-demand entity creating using dynamic memory when required.
+This dynamic memory usage uses micro-ROS allocators.
+
+- ``RMW_UXRCE_ALLOW_DYNAMIC_ALLOCATIONS``: Allow dynamic memory allocations when creating micro-ROS entities. Default value: ``OFF``.
+
+Middleware related memory
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By means of the following build flags, the user can configure the behavior of communication buffers:
+
+- ``RMW_UXRCE_STREAM_HISTORY``: Maximum number of buffers of size ``UCLIENT_[XXX]_TRANSPORT_MTU`` that the XRCE-DDS layer is provided with. **It shall be power of 2**. Default value: 4.
+- ``RMW_UXRCE_STREAM_HISTORY_INPUT``: Override for ``RMW_UXRCE_STREAM_HISTORY`` for input buffers. Default value: ``RMW_UXRCE_STREAM_HISTORY``.
+- ``RMW_UXRCE_STREAM_HISTORY_OUTPUT``: Override for ``RMW_UXRCE_STREAM_HISTORY`` for output buffers. Default value: ``RMW_UXRCE_STREAM_HISTORY``.
+- ``RMW_UXRCE_MAX_HISTORY``: Maximum number of slots for storing incoming data between ``wait()`` and ``take()`` operations at RMW layer. It size is ``UCLIENT_[XXX]_TRANSPORT_MTU * RMW_UXRCE_STREAM_HISTORY[_INPUT]``  Default value: 8.
+
+.. note::
+
+    When using Best Effort entities, the maximum serialized size of a topic shall fit in ``UCLIENT_[XXX]_TRANSPORT_MTU`` both for outgoing and incoming data.
+
+    When using Reliable entities, the maximum serialized size of a topic shall fit in ``UCLIENT_[XXX]_TRANSPORT_MTU * RMW_UXRCE_STREAM_HISTORY`` both for outgoing and incoming data.
+
+    When receiving data, and all ``RMW_UXRCE_MAX_HISTORY`` are occupied, the reception entity behavior is determined by History QoS.
+
+Entity Names
+^^^^^^^^^^^^
+
+Regarding the name of entities, topics and types, the user can configure the maximum length of the name by means of the following flags:
+
+- ``RMW_UXRCE_NODE_NAME_MAX_LENGTH``: Maximum number of characters for a node name. Default value: 60.
+- ``RMW_UXRCE_TOPIC_NAME_MAX_LENGTH``: Maximum number of characters for a topic name. Default value: 60.
+- ``RMW_UXRCE_TYPE_NAME_MAX_LENGTH``: Maximum number of characters for a type name. Default value: 100.
+
+
+.. _tutorials_micro_memory_management_message_memory:
+
+Message memory
+--------------
+
+Message memory handling is an important part of the micro-ROS Client memory handling due to the fact that **micro-ROS Client does not initialized by default the type memory**.
+This means that the user must initialize the type memory before using it.
+This consideration needs to be taken into account both for outgoing and incoming messages.
+
+.. warning::
+
+    micro-ROS provides an API for initializing the type memory that can be found in :ref:`micro-ROS Types Utilities<tutorials_micro_utilities_type_utilities>`.
```

Comment:
I think `Note` suits better here than `warning`

Reply:
Agree... but I want users to understand that this section is for understanding purposes and that they have the API. I want more than a note but less than a warning

Reply:
Fixed here: 9463bbfeeccd4acfd5fba0677ba74ec79a2782d3